unstructured_ingest/v2/processes/connectors/discord.py

```diff
@@ -0,0 +1,169 @@
+
+CONNECTOR_TYPE = "discord"
+
+
+class DiscordAccessConfig(AccessConfig):
+    token: str = Field(description="Discord API token")
+
+
+class DiscordConnectionConfig(ConnectionConfig):
+    access_config: Secret[DiscordAccessConfig]
+    channels: Optional[List[str]] = Field(
+        default=None, description="List of Discord channel IDs to process"
+    )
+
+
+class DiscordIndexerConfig(IndexerConfig):
+    recursive: bool = Field(
+        default=False, description="Recursively process child channels and messages"
+    )
+
+
+@dataclass
+class DiscordIndexer(Indexer):
+    connection_config: DiscordConnectionConfig
+    index_config: DiscordIndexerConfig
+
+    @requires_dependencies(["discord"], extras="discord")
+    def get_client(self):
+        import discord
+
+        intents = discord.Intents.default()
+        intents.message_content = True
+        return discord.Client(intents=intents)
+
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        client = self.get_client()
+        processed_channels: Set[str] = set()
+        channels_to_process: Set[str] = set(self.connection_config.channels or [])
+
+        while channels_to_process:
+            for channel_id in list(channels_to_process):
+                if channel_id in processed_channels:
+                    continue
+
+                processed_channels.add(channel_id)
+                channels_to_process.remove(channel_id)
+                file_data = self.get_channel_file_data(channel_id=channel_id, client=client)
+                if file_data:
+                    yield file_data
+
+    @requires_dependencies(["discord"], extras="discord")
+    def get_channel_file_data(self, channel_id: str, client) -> Optional[FileData]:
+        try:
+            # Fetch channel metadata
+            channel_metadata = {}  # Replace with actual metadata retrieval
+            date_created = dt.datetime.utcnow().isoformat()
+            identifier = channel_id
+            source_identifiers = SourceIdentifiers(
+                filename=f"{channel_id}.txt",
+                fullpath=channel_id,
+                rel_path=channel_id,
+            )
+            metadata = FileDataSourceMetadata(
+                date_created=date_created,
+                date_modified=date_created,
+                record_locator={"channel_id": channel_id},
+                date_processed=str(dt.datetime.utcnow().timestamp()),
+            )
+            additional_metadata = channel_metadata
+            return FileData(
+                identifier=identifier,
+                connector_type=CONNECTOR_TYPE,
+                source_identifiers=source_identifiers,
+                metadata=metadata,
+                additional_metadata=additional_metadata,
+            )
+        except Exception as e:
+            logger.error(f"Error retrieving channel {channel_id}: {e}")
+            return None
```

Comment:
Rather than catch any error quietly, lets not catch it and have the downloader raise the error. I don't want records to be dropped in case something else breaks in this chunk of code.